Purchases/Logging/Strings/CodableStrings.swift

```diff
@@ -16,13 +16,33 @@ import Foundation
 // swiftlint:disable identifier_name
 enum CodableStrings {
 
-    static func unexpectedValueError(for type: Any.Type) -> String {
-        "Found unexpected value for type: \(type)"
-    }
-    static func valueNotFoundError(for type: Any.Type) -> String {
-        "No value found for type: \(type)"
-    }
-    static let invalid_json_error: String = "The given data was not valid JSON\n%@"
-    static let decoding_error = "Couldn't decode data from json\n%@"
+    case unexpectedValueError(type: Any.Type)
+
+    case valueNotFoundError(type: Any.Type)
+
+    case invalid_json_error(jsonData: [String: Any])
+
+    case decoding_error(errorMessage: String)
+
+}
 
+extension CodableStrings: CustomStringConvertible {
+
+    var description: String {
+        switch self {
+
+        case .unexpectedValueError(let type):
+            return "Found unexpected value for type: \(type)"
+
+        case .valueNotFoundError(let type):
+            return "No value found for type: \(type)"
+
+        case .invalid_json_error(let jsonData):
+            return "The given json data was not valid: \n\(jsonData)"
+
+        case .decoding_error(let errorMessage):
+            return "Couldn't decode data from json. Error: \n\(errorMessage)"
+
+        }
+    }
```

Comment:
Style: needs extra return

Reply:
do we have a rule for a line break after the end of a switch? this closing bracket is closing the method, not the extension body

Reply:
happy to update it if you like it better

Reply:
Whoops, no, I thought it was closing the body.